tests/apollo/util/skvbc.py

```diff
@@ -493,90 +472,113 @@ async def run_concurrent_ops(self, num_ops, write_weight=.70):
 
-    async def send_indefinite_tracked_batch_writes(self, batch_size, time_interval=.01):
+    async def send_indefinite_batch_writes(self, batch_size, time_interval=.01):
         max_size = len(self.keys) // 2
         while True:
             client = self.bft_network.random_client()
             async with trio.open_nursery() as nursery:
                 try:
-                    nursery.start_soon(self.send_tracked_write_batch, client, max_size, batch_size)
+                    nursery.start_soon(self.send_write_kv_set_batch, client, max_size, batch_size)
                 except:
                     pass
                 await trio.sleep(time_interval)
 
-    async def send_indefinite_tracked_ops(self, write_weight=.70, time_interval=.01):
+    async def send_indefinite_ops(self, write_weight=.70, time_interval=.01):
         max_size = len(self.keys) // 2
         while True:
             client = self.bft_network.random_client()
             async with trio.open_nursery() as nursery:
                 try:
                     if random.random() < write_weight:
-                        nursery.start_soon(self.send_tracked_write, client, max_size)
+                        nursery.start_soon(self.send_write_kv_set, client, None, max_size, False, False, False)
                     else:
-                        nursery.start_soon(self.send_tracked_read, client, max_size)
+                        nursery.start_soon(self.send_read_kv_set, client, None, max_size)
                 except:
                     pass
                 await trio.sleep(time_interval)
 
-    async def send_tracked_write_batch(self, client, max_set_size, batch_size, read_version = None, long_exec = False):
+    async def send_write_kv_set_batch(self, client, max_set_size, batch_size, read_version = None, long_exec = False):
         msg_batch = []
         batch_seq_nums = []
         client_id = client.client_id
         if read_version is None:
-            read_version = self.tracker.read_block_id()
+            read_version=0
+            if self.tracker is not None:
+                read_version = self.tracker.read_block_id()
         for i in range(batch_size):
-            max_read_set_size = 0 if self.tracker.no_conflicts else max_set_size
-            readset = self.tracker.readset(0, max_read_set_size)
-            writeset = self.tracker.writeset(max_set_size)
+            max_read_set_size = max_set_size
+            if self.tracker is not None:
+                max_read_set_size = 0 if self.tracker.no_conflicts else max_set_size
+            readset = self.readset(0, max_read_set_size)
+            writeset = self.writeset(max_set_size)
             msg_batch.append(self.write_req(readset, writeset, read_version, long_exec))
             seq_num = client.req_seq_num.next()
             batch_seq_nums.append(seq_num)
-            self.tracker.send_write(client_id, seq_num, readset, dict(writeset), read_version)
+            if self.tracker is not None:
```

Comment:
Multiple checks for self.tracker can be avoided as the whole code block process holds valid only if tracker is enabled.

Reply:
Thanks Celine, will do that